test/unit/plugins/validate-semantic/oas3/refs.js

```diff
@@ -206,4 +206,127 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
-})
+
+  describe("parameter $refs should not point to header components", () => {
+    it("should return an error when a parameter incorrectly references a response header component", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            parameters: [
+              {
+                $ref: "#/components/headers/foo"
+              }
+            ],
+            get: {
+              parameters: [
+                {
+                  $ref: "#/components/headers/foo"
+                }
+              ],
+              responses: {
+                "200": {
+                  description: "OK"
+                }
+              }
+            }
+          }
+        },
+        components: {
+          parameters: {
+            myParam: {
+              $ref: "#/components/headers/foo"
+            }
+          }
```

Comment:
Suggestion: add a dummy header component named `foo` to avoid extra `$ref` resolution errors in this test
```suggestion
          },
          headers: {
            foo: {}
          }
```

Reply:
Updated! This improved the test case.